react/ProductLink.tsx

```diff
@@ -68,13 +69,21 @@ function ProductLink(props: Props) {
     [classes.label]: displayMode === 'button',
   })
 
+  const handlePrevent = (e: React.MouseEvent) =>{
+    e.stopPropagation()
+    e.nativeEvent.stopImmediatePropagation()
+
+  }
+
   return (
-    <Link target={target} to={resolvedLink} className={rootClasses} rel={rel}>
-      {label && <span className={labelClasses}>{label}</span>}
-      {hasChildren(children) && displayMode === 'anchor' && (
-        <div className={handles.childrenContainer}>{children}</div>
-      )}
-    </Link>
+    <div onClick={handlePrevent} className={handles.linkContainer}>
+      <Link target={target} to={resolvedLink} className={rootClasses} rel={rel}  onClick={handlePrevent}>
+        {label && <span className={labelClasses}>{label}</span>}
+        {hasChildren(children) && displayMode === 'anchor' && (
+          <div className={handles.childrenContainer}>{children}</div>
+        )}
+      </Link>
+    </div>
```

Comment:
You don't need a Container link as you are already using StopPropagation on Link
```suggestion
    <Link
      target={target}
      to={resolvedLink}
      className={rootClasses}
      rel={rel}
      onClick={handlePrevent}
    >
      {label && <span className={labelClasses}>{label}</span>}
      {hasChildren(children) && displayMode === 'anchor' && (
        <div className={handles.childrenContainer}>{children}</div>
      )}
    </Link>
```

Reply:
@emersonlaurentino I don't understand exactly what do you suggest, can you explain to me? thank you